### PR TITLE
v2: redis proxy filter

### PR DESCRIPTION
**Pull request by rshriram**
*title*: convert redis proxy filter to v2 format

*Description*: Redis filter in v2 format, in addition to some cleanups in mongo filter.
*Risk Level*: Low
*Testing*: existing unit and integration tests that use v1 format, in addition to v2 format tests.

Signed-off-by: Shriram Rajagopalan <shriram@us.ibm.com>

### CODE REVIEW

File: source/server/config/network/mongo_proxy.cc [hunk 1]
```diff
@@ -8,28 +8,28 @@
...
 namespace Configuration {
 
 NetworkFilterFactoryCb MongoProxyFilterConfigFactory::createMongoProxyFactory(
-    const envoy::api::v2::filter::network::MongoProxy& mongo_proxy, FactoryContext& context) {
+    const envoy::api::v2::filter::network::MongoProxy& config, FactoryContext& context) {
 
-  ASSERT(!mongo_proxy.stat_prefix().empty());
-  std::string stat_prefix = "mongo." + mongo_proxy.stat_prefix() + ".";
+  ASSERT(!config.stat_prefix().empty());
+  std::string stat_prefix = fmt::format("mongo.{}.", config.stat_prefix());
```
Comment (mattklein123) [inline source/server/config/network/mongo_proxy.cc]: nit: const

File: source/server/config/network/redis_proxy.cc [hunk 2]
```diff
@@ -14,16 +15,20 @@ namespace Envoy {
...
-RedisProxyFilterConfigFactory::createFilterFactory(const Json::Object& config,
-                                                   FactoryContext& context) {
+NetworkFilterFactoryCb RedisProxyFilterConfigFactory::createRedisProxyFactory(
+    const envoy::api::v2::filter::network::RedisProxy& config, FactoryContext& context) {
+
+  ASSERT(!config.stat_prefix().empty());
+  ASSERT(!config.cluster().empty());
+  ASSERT(config.has_settings());
+
+  std::string stat_prefix = fmt::format("redis.{}.", config.stat_prefix());
```
Comment (mattklein123) [inline source/server/config/network/redis_proxy.cc]: nit: const

**Comment by rshriram** [comment]
I had to force push due to some issues. Fixed the const.

File: test/common/redis/proxy_filter_test.cc [hunk 3]
```diff
@@ -272,7 +284,7 @@ TEST_F(RedisProxyFilterTest, ProtocolError) {
   EXPECT_CALL(filter_callbacks_.connection_, close(Network::ConnectionCloseType::NoFlush));
   EXPECT_EQ(Network::FilterStatus::StopIteration, filter_->onData(fake_data));
 
-  EXPECT_EQ(1UL, store_.counter("redis.foo.downstream_cx_protocol_error").value());
+  EXPECT_EQ(1UL, config_->stats_.downstream_cx_protocol_error_.value());
```
Comment (rshriram) [inline test/common/redis/proxy_filter_test.cc]: @mattklein123 I had to fix this in order for the proxy filter test to pass. I am not sure if the changes I did caused some other regression. The stats prefix seems to have been set properly.

Comment (rshriram) [reply test/common/redis/proxy_filter_test.cc]: nm.. fixed the bug..

**Pull request merged**